holoviews/plotting/bokeh/element.py

```diff
@@ -168,18 +166,56 @@ def __init__(self, element, plot=None, **params):
         self.current_ranges = None
         super(ElementPlot, self).__init__(element, **params)
         self.handles = {} if plot is None else self.handles['plot']
-        element_ids = self.hmap.traverse(lambda x: id(x), [Element])
-        self.static = len(set(element_ids)) == 1 and len(self.keys) == len(self.hmap)
+        self.static = len(self.hmap) == 1 and len(self.keys) == len(self.hmap)
+        self.callbacks = self._init_callbacks()
 
 
-    def _init_tools(self, element):
+    def _init_callbacks(self):
+        """
+        Initializes any callbacks for streams which have defined
+        the plotted object as a source.
+        """
+        if not self.static or isinstance(self.hmap, DynamicMap):
+            source = self.hmap
+        else:
+            source = self.hmap.last
+        streams = Stream.registry.get(source, [])
+        registry = Stream._callbacks['bokeh']
```

Comment:
Ah, so ``_callbacks`` contains the backend specific code? Worth clarifying where ``_callbacks`` is defined on ``Stream`` (see comment below).